core/arch/arm/include/kernel/thread.h

```diff
@@ -308,6 +323,9 @@ void thread_restore_irq(void);
 #define THREAD_EXCP_ABT	(1 << 2)
 #define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
 
+/* Defines the bit mask for the foreign interrupts */
+#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
```

Comment:
Yet, @davwan01, I agree with your comment.

In `thread_set/get_exceptions()`, the numerical values of `THREAD_FOREIGN/NATIVE_INTR` are used both as IDs for the arguments of the `thread_xxx()` routines and for the `CPSR`/`DAIF` registers bit fields. This will not match the `CPSR`/`DAIF` registers layout and the several GICs versions. 
* either define `THREAD_FOREIGN/NATIVE_INTR` according to the GIC version 
* or translate `THREAD_EXCP_FOREIGN/NATIVE_INTR` into a `CPSR/DAIF` mask, 
* or don't define them and define only a THREAD_EXCP_ALL_INTR (as suggested by Jens).